web_src/js/features/repo-diff.ts

```diff
@@ -19,15 +19,35 @@ function initRepoDiffReviewButton() {
   const counter = reviewBox.querySelector('.review-comments-counter');
   if (!counter) return;
 
-  $(document).on('click', 'button[name="pending_review"]', (e) => {
-    const $form = $(e.target).closest('form');
-    // Watch for the form's submit event.
-    $form.on('submit', () => {
-      const num = parseInt(counter.getAttribute('data-pending-comment-number')) + 1 || 1;
+  function handleFormSubmit($form, $textarea) {
+    $form.one('submit', (event) => {
+      if ($textarea.val().trim() === '') {
+        event.preventDefault();
+        return;
+      }
+      const num = (parseInt(counter.getAttribute('data-pending-comment-number')) || 0) + 1;
       counter.setAttribute('data-pending-comment-number', num);
       counter.textContent = num;
       animateOnce(reviewBox, 'pulse-1p5-200');
     });
+  }
+
+  // Handle submit on click
+  $(document).on('click', 'button[name="pending_review"]', (e) => {
+    const $form = $(e.target).closest('form');
+    const $textarea = $form.find('textarea');
+    handleFormSubmit($form, $textarea);
+    $form.trigger('submit');
```

Comment:
Explicitly triggering submission here seems to work, despite it not being included before. The only thing is explicitly submitting on each click requires us to do form validation, which I'd like to avoid.

I have no clue why, but before, the on submit handler was being called even if the textarea was empty (i.e. on click event would attach the handler and also somehow 'submit' to run the handler)... this would cause the counter to increment even if a comment is not posted